# Python program to reverse a number with explanation

321
if  we reverse this no 
 123

In [31]:
n=int(input(" Enter a  number :")) # 1234 # n%10 = remainder 
print(" before  reversing a no :", n)
r=0 # reverse number - store a value  - counter variable 123%10 =  3 
while n!=0:# n=1- is not equal to 0 
    r= r * 10 + n%10 # r= 432 * 10 + 1 = 4320+1= 4321
    n=n//10 # 12//10 = 1.2= ignore point value= 1
print("reverse no:", r)


 Enter a  number :121
 before  reversing a no : 121
reverse no: 121


In [2]:
n=int(input(" Enter a  number :")) # 123 # n%10 = remainder 
print(" before  reversing a no :", n)
s= int(str(n)[::-1])
print("reverse no :", s)

 Enter a  number :123
 before  reversing a no : 123
reverse no : 321


In [3]:
num = int(input("Enter a number :"))
temp= num
count=0

while temp!=0:
    digit = temp%10
    temp //=10
    print(digit,end="")
    count +=1

Enter a number :123
321

# Armstrong number program

- It is a number that is equal to the sum of its own digits raised to the power of the number of digits.
- For eg: 153, is an Armstrong number because it has 3 digits, and 1^3 + 5^3 + 3^3 = 1 + 125 + 27 = 153

Lets see it practically by calculating it,

Suppose I am taking 153 for an example 

First calculate the cube of its each digits

  1^3 = (1*1*1) = 1

  5^3 = (5*5*5) = 125

  3^3= (3*3*3) = 27
  
  
  Now add the cube 

  1+125+27 = 153


It means 153 is an Armstrong number

- 153 is an Armstrong number because:

   1^3 + 5^3 + 3^3 = 153


- 371 is an Armstrong number because:

   3^3 + 7^3 + 1^3 = 371
- 407 is an Armstrong number because:

    4^3 + 0^3 + 7^3 = 407
- 1634 is an Armstrong number because:

    1^4 + 6^4 + 3^4 + 4^4 = 1634
- 8208 is an Armstrong number because:

     8^4 + 2^4 + 0^4 + 8^4 = 8208



In [17]:
num=int(input(" Enter  a number :")) #153 
s=0 # sum 
temp=num
count=len(str(num)) # 3  # len()- length of string , list tuple or array
while temp>0: # 1 > 0 
    digit=temp%10  # 1 % 10= 1 
    s+=digit**count # s= 27 + 1**  3 = 27+125 = 152 +1 =153
    temp=temp//10 # 15//10=1.5=1
print("sum ",s)
if num==s: # 153== s(153)
    print(num,"given no  is armstrong no")
else:
    print(num,"give no is not armstrong no ")


 Enter  a number :123
sum  36
123 give no is not armstrong no 


In [18]:
1**3 + 5** 3+ 3**3
print(1**3+2**3+3**3)

36


# Prime Number Program

In [26]:
# Whether a number is a prime no or not  
n=int(input(" Enter a number:")) # 7 - 1 or 7- c= 2- counter 
c=0
i=1
while i<=n: # 8 <= 7
    if n % i ==0: # 7 % 7= 0 
        c+=1# c=c+1=1+1= 2 # c=2
    i=i+1# i=7+1=8

if c==2:
    print(n,"is a prime no")
else:
    print(n,"is not  a prime no")

 Enter a number:5
5 is a prime no


In [25]:
n = int(input("please give a number : "))#13
i,temp=0,0
for i in range(2,n//2): # (2 to 6) 
    if n%i == 0:# 13%2!=0
        temp=1
        break
if temp == 1:
    print("given number is not prime")
else:
    print("given number is prime")


please give a number : 13
given number is prime


6

# Fibonacci series program

A Fibonacci series is a series in which next number is a sum of previous two numbers

For example : 0, 1, 1, 2, 3, 5, 8 ……

In Fibonacci Series, first number starts with 0 and second is with 1 and then its grow like,

0

1

0 + 1 = 1


1 + 1 = 2

1 + 2 = 3

2 + 3 = 5 and so on

In [30]:
n=int(input("enter a no:")) #5 
first,second= 0,1
for i in range(0,n):#(0,4)
    if i<=1:# 1<=1 # i=0,1 
        result=i # 0, 1
    else: # i= 2 To 4  i=4
        result=first+second # 1+2=3
        first=second # 1=2 =1
        second=result # 1=3
    print(result)
        

enter a no:5
0
1
1
2
3


# Python program for palindrome 

In [36]:
n=int(input(" Enter a  number :"))
temp=n
print(" before  reversing a no :", n)
r=0 
while n!=0:
    r= r * 10 + n%10 
    n=n//10 
print("reverse no:", r)
if r==temp:
    print("palindrome ")
else:
    print("not palindrome")

 Enter a  number :111
 before  reversing a no : 111
reverse no: 111
palindrome 


A Palindrome number is a number which reverse is equal to the original number means number itself.


For example : 121, 111, 1223221, etc.

In the above example you can see that 121 is a palindrome number. Because reverse of the 121 is same as 121

# Perfect number

A perfect number is a positive integer that is equal to the sum of its positive divisors, excluding the number itself.

Lets understand it with example

6 is a positive number and its divisor is 1,2,3 and 6 itself.

In [39]:
n=int(input(" enter a number:")) # 6
s=0 # counter
for i in range(1,(n//2)+1):# (1, 4)
    rem=n%i # 6 % 3! =0 # remainder
    if rem==0:# 0==0
        s=s+i #s=3+3=6
print("sum of divisors:", s)
if s==n: # 6==6
    print("perfect no")
else:
    print("not perfect no ") 

 enter a number:7
sum of divisors: 1
not perfect no 


But we should not include 6 as by the definition of perfect number.

Lets add its divisor excluding itself

1+2+3 = 6 which is equal to number itself.

# Program for Sum the digits of a given number

Given a number and the task is to find sum of digits of this number in Python. 
Examples: 
 

- Input : n = 87 
    Output : 15 
- Input : n = 111 
    Output : 3

In [43]:
# 52= 5+2= 7
n=int(input("Enter a number:"))
s=0 # - store a sum of digits of a number
for i  in str(n):
    print(i)# i=4
    s=s+int(i)# s=0+i=1+4=5
print("sum of a digits of a number:",s)

Enter a number:14
1
4
sum of a digits of a number: 5


In [49]:
num = int(input()) # 25
prod = 1
temp = num
while num!=0:# 2!=0
    digit = num%10  # 2 % 10 
    prod = prod*digit # ( 5 * 2 )=10
    num //=10 #  25//10=2 
print(prod)
str1="python"
print(str1[::-1])

25
10
nohtyp


# Product of  digits in an given number

In [ ]:
 # To find sum of squares of a digits in a number
# to findsum of cubes of a digits in a number
# write a program to find sum of even and odd digits in a given number.
# factorial of a given number